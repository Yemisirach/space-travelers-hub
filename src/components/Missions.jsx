import React, { Component } from "react";

export default class Missions extends Component {
  render() {
    return <div></div>;
  }
}
