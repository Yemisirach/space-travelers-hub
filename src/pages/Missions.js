import { useSelector } from 'react-redux';
import Mission from '../components/Mission/Mission';

const Missions = () => {
  const { missionList } = useSelector((state) => state.missions);

  return (
    <>
      <div className="missions-info">
        <p>Mission</p>
        <p>Description</p>
        <p>Status</p>
        <p />
      </div>
      <div>
        {missionList.map((mission) => (
          <Mission
            key={mission.id}
            id={mission.id}
            name={mission.name}
            description={mission.description}
            reserved={mission.reserved}
          />
        ))}
      </div>
    </>
  );
};

export default Missions;
