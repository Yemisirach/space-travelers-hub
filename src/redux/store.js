import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './mission/missionSlice';
import rocketReducer from './Rocket/rocketSlice';
import dragonReducer from './dragon/dragonSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
    dragons: dragonReducer,
  },
});

export default store;
