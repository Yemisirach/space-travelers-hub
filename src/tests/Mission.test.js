import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Missions from '../components/Mission/Mission';
import store from '../redux/store';

describe('Missions', () => {
  test('Render Missions Component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Missions />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
