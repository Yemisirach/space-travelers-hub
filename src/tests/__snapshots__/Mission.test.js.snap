// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Render Missions Component 1`] = `
<div>
  <section
    className="missions"
  >
    <div
      className="missions-details"
    >
      <h3 />
      <p />
      <div
        className="member-sign"
      >
        <h4
          className="status notActive"
        >
          NOT A MEMBER
        </h4>
      </div>
      <div
        className="btn-container"
      >
        <button
          className="join-btn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </div>
    </div>
  </section>
</div>
`;
